lib/engine/game/g_18_india/step/gauge_change_border.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/tracker'
+
+module Engine
+  module Game
+    module G18India
+      module Step
+        module GaugeChangeBorder
+          BORDER_TYPE_TO_CHANGE = :province
+          STATUS_TO_DISABLE_CHANGE = 'no_gauge_change'
+          GAGUE_CHANGE_COLOR = :red
+
+          # modified to prevent removal of province borders that had a cost (the cost is from a water border)
+          def remove_border_calculate_cost!(tile, entity_or_entities, spender)
+            # entity_or_entities is an array when combining private company abilities
+            entities = Array(entity_or_entities)
+            entity, *_combo_entities = entities
+
+            hex = tile.hex
+            types = []
+            LOGGER.debug 'GaugeChangeBorder::remove_border_calculate_cost Called >>'
+
+            total_cost = tile.borders.dup.sum do |border|
+              cost = border.cost || 0
+              edge = border.edge
+              neighbor = hex.neighbors[edge]
+              next 0 if !hex.targeting?(neighbor) || !neighbor.targeting?(hex)
+
+              types << border.type
+              if border.type == BORDER_TYPE_TO_CHANGE
+                # Remove the existing border on tile and neighbor
+                tile.borders.delete(border)
+                neighbor.tile.borders.map! { |nb| nb.edge == hex.invert(edge) ? nil : nb }.compact!
```

Comment:
Is this the same as:
```suggestion
                neighbor.tile.borders.reject! { |nb| nb.edge == hex.invert(edge) }
```
Using `map!` to change a value to nil and then remove it with `compact!` feels a bit strange.

Reply:
NOTE: This is in the existing code in `Engine::Step::Tracker` for `remove_border_calculate_cost!`.

I will test the change to see if there is a reason the existing cost wrote it this way.

Reply:
Tested change, it works as expected.  Not sure why it was originally coded the other way.